Update database with current course data

miclaymon opened the pull request:
Database now should load with accurate course information for the selected courses, including sections, professor EUIDs, and seat numbers. This should not affect the functionality of the application, only the data.

You may want to modify the `CourseOutcome.LinkToMajorOutcome(...)` lines before approving, because I am not sure how we are supposed to map those.
Currently, I left the previous teams' test data in including the outcomes like "make gud at computers", and courses like CSCE 3600. We can have that removed if you want.

Let me know if I need to make any changes, or you can make them yourself if you want to.

miclaymon committed:
Populate database with Fall 2022 courses.

miclaymon committed:
Add number of seats for each section

Replaced missing number of students for each section with the correct number of seats available for each section.

miclaymon committed:
Minor update to README

Set the declared project semester to the current one (Fall 2022).

miclaymon commented on AbetApi/Data/Database.cs:
```diff
@@ -43,120 +43,248 @@ await using (var context = new ABETDBContext())
+            _ = CourseOutcome.CreateCourseOutcome("Fall", 2022, "CSCE", "2110", new CourseOutcome("8", "Use of hash tables in design of software."));
+
+            // link major outcomes to course outcomes
+            // HELP: how do you guys want to link these? do we need to ask for more info?
```
Please take a look at the `CourseOutcome.LinkToMajorOutcome(...)` and let me know how you want to proceed.

miclaymon commented:
Even though the code to add the data to the database is present it does not seem to be replacing the existing Spring 2022 data and the Fall 2022 semester does not show up in the dropdown.

@Nathanlenk can you take a look and let me know why the data isn't appearing?

miclaymon committed:
Comment Spring 2022 data so that only Fall 2022 is data is read.

Nathanlenk commented:
So we need to use the custom controller to update the database.

![Untitled](https://user-images.githubusercontent.com/90060370/204154551-71a2ff0d-8de9-46b8-a493-17f775b1ce7e.png)

Click the /Custom/WipeDBTables and then execute to clear out the old junk. 
Then click /Custom and execute to fill in the new data.

Thanks for filling all this!

miclaymon committed:
WipeTables()

miclaymon commented:
Nathan confirmed that after running the endpoints `/Custom/WipeDBTables` and `/Custom`, the data now successfully loads.
I have also successfully got the new data to load using this method.

miclaymon committed:
Update README.md

Nathanlenk committed:
Added User Data

- Add roles
- Add instructor user information
- Add roles to the instructors

miclaymon committed:
Fix typo

Change "Spring" to "Fall" when linking major outcomes.

dylanwindebank committed:
Update README.md

Added a section for MySQL installation and connecting it to the back end.

Nathanlenk committed:
Minor Bug fix

Front end was having instructor use the getsection controller. Needed to make it accessible by instructors.

Nathanlenk committed:
Coordinator Update

Added the testing coordinator login to the database as a EUID.

Can now login as the coordinator and test those functions with all of the data.

Nathanlenk committed:
CE outcome links

Added them

Nathanlenk committed:
Small Edit

Forgot to change data I copy/pasted.

huynggg committed:
CS Outcome Links

Commit:
added IT connecting student outcome to course outcome

dylanwindebank committed:
Update README.md

fixed years

miclaymon committed:
Remove unused duplicate courses

Some of the `Course.AddCourse(...)` calls were commented out because they were typically identical to the lines preceding them. The reason why they were originally added if from the spreadsheet of class data: the spreadsheet will, for each _section_ generate the lines of code needed to create the _course_ and _section_. This meant that courses with multiple sections had code generated to create the course again for each additional section. When the code was copied into the database, it was commented out as a duplicate.
It has now been removed, since it did not serve a purpose.

Nathanlenk committed:
Quick Fix for Major Link

There was a check that was not allowing multiple student outcome links to the same course outcome. This was commented out and may need to be reimplemented differently later.

miclaymon committed:
Update README.md

Fix broken link for ".NET"
Add link to download MySQL Worbench
